tests/integration/aic.rs

```diff
@@ -0,0 +1,242 @@
+use cjms::{
+    controllers::aic::AICResponse,
+    models::aic::{AICModel, AIC},
+};
+use serde_json::json;
+use time::OffsetDateTime;
+use uuid::{Uuid, Version};
+
+use crate::utils::{
+    random_ascii_string, send_get_request, send_post_request, send_put_request, spawn_app, TestApp,
+};
+
+#[tokio::test]
+async fn test_aic_get_is_not_allowed() {
+    let app = spawn_app().await;
+    let test_cases = vec!["/aic", "/aic/123"];
+    for path in test_cases {
+        let r = send_get_request(&app, path).await;
+        assert_eq!(r.status(), 405, "Failed on path: {}", path);
+    }
+}
+
+#[tokio::test]
+async fn aic_create_with_bad_data() {
+    let app = spawn_app().await;
+    let test_cases = [
```

Comment:
Just a note that I came across this [this library](https://crates.io/crates/rstest) when Googling around to see if pytest-style parametrized tests were possible in Rust. This lib implements fixtures and parametrization which seems cool. It even gets a mention in the zero2prod book, though the author says that it's not compatible with the `tokio::test` macro. Although the lib seems to have had some development since the book's release and the README describes what appears to be a simple integration that actually does work.

Anyway. Not related to this PR really and may not be relevant or useful to us, but I thought it was at least worth sharing.